### ISSUE TITLE
[2.0.4 bug] Cannot use a scalar value as an array

### ISSUE THREAD

**Issue by galki** [phalcon/cphalcon#10648]
I'm using PHP 5.6.8 and getting the below notices when I have error reporting turned on.
It looks like the way an array is written as another array's key is the cause.

// Notice: Cannot use a scalar value as an array in phalcon/mvc/model/resultset/simple.zep on line 209
```let renamed[renamedKey[0]] = value;```

// Notice: Cannot use a scalar value as an array in phalcon/mvc/model.zep on line 3600
```let snapshot[attribute[0]] = value;```

**Comment by olivier-monaco** [comment]
@andresgutierrez look here: https://github.com/phalcon/cphalcon/blob/master/phalcon/mvc/model/resultset/simple.zep#L205

<pre>
		                    if !fetch renamedKey, renamedKey[0] {
                	            throw new Exception("Column '" . key . "' is not part of the column map");
                        	}

							let renamed[renamedKey[0]] = value;
</pre>

We replace renamedKey by its first item but them ask for the first item of renamedKey...

It must be
<pre>
let renamed[renamedKey] = value;
</pre>

Same for the other notice.

**Comment by olivier-monaco** [comment]
Usecase:
```php
<?php
$di = new Phalcon\DI();

$di['db'] = new Phalcon\Db\Adapter\Pdo\MySQL([
    "host"     => "localhost",
    "username" => "phalcon_test",
    "password" => "phalcon_test",
    "dbname"   => "phalcon_test",
]);

\Phalcon\Mvc\Model::setup(['castOnHydrate' => true]);

$di['modelsManager'] = new Phalcon\Mvc\Model\Manager();
$di['modelsManager']->setDi($di);
$di['modelsMetadata'] = new Phalcon\Mvc\Model\MetaData\Memory();

class Robot extends Phalcon\Mvc\Model
{
    public $id;
}

$q = new Phalcon\Mvc\Model\Query('select * from Robot limit 1', $di);
$r = $q->execute();
var_dump($r->toArray());
```

Result:
```
PHP Notice:  Cannot use a scalar value as an array in phalcon/mvc/model/resultset/simple.zep on line 209 in /volumes/data1/vhosts-dev/libraries/cphalcon-olivier/issues/i10673a.php on line 24
PHP Stack trace:
PHP   1. {main}() /volumes/data1/vhosts-dev/libraries/cphalcon-olivier/issues/i10673a.php:0
PHP   2. Phalcon\Mvc\Model\Resultset\Simple->toArray() /volumes/data1/vhosts-dev/libraries/cphalcon-olivier/issues/i10673a.php:24

Notice: Cannot use a scalar value as an array in phalcon/mvc/model/resultset/simple.zep on line 209 in /volumes/data1/vhosts-dev/libraries/cphalcon-olivier/issues/i10673a.php on line 24

Call Stack:
    0.0001     263952   1. {main}() /volumes/data1/vhosts-dev/libraries/cphalcon-olivier/issues/i10673a.php:0
    0.0021     287024   2. Phalcon\Mvc\Model\Resultset\Simple->toArray() /volumes/data1/vhosts-dev/libraries/cphalcon-olivier/issues/i10673a.php:24

array(1) {
  [0] =>
  array(1) {
    '' =>
    string(1) "1"
  }
}
```

**Comment by andresgutierrez** [comment]
Fixed in the 2.0.x branch
